vexim/sitechangesubmit.php

```diff
@@ -1,99 +1,145 @@
 <?php
-  include_once dirname(__FILE__) . "/config/variables.php";
-  include_once dirname(__FILE__) . "/config/authsite.php";
-  include_once dirname(__FILE__) . "/config/functions.php";
-  include_once dirname(__FILE__) . "/config/httpheaders.php";
+include_once dirname(__FILE__) . "/config/variables.php";
+include_once dirname(__FILE__) . "/config/authsite.php";
+include_once dirname(__FILE__) . "/config/functions.php";
+include_once dirname(__FILE__) . "/config/httpheaders.php";
 
-  $success = false;
-  if (isset($_POST['sadisable'])) {$action = 'sadisable';}
-  if (isset($_POST['avdisable'])) {$action = 'avdisable';}
-  if (isset($_POST['avscan'])) {$_POST['avscan'] = 1;} else {$_POST['avscan'] = 0;}
-  if (isset($_POST['spamassassin'])) {$_POST['spamassassin'] = 1;} else {$_POST['spamassassin'] = 0;}
-  if (isset($_POST['enabled'])) {$_POST['enabled'] = 1;} else {$_POST['enabled'] = 0;}
-  if (isset($_POST['pipe'])) {$_POST['pipe'] = 1;} else {$_POST['pipe'] = 0;}
-  if ($_POST['max_accounts'] == '') {$_POST['max_accounts'] = '0';}
-  if (isset($_POST['clear'])) {
+$success = false;
+
+// Detect the user action
+if (isset($_POST['allusers'])) {
+  $action = 'allusers';
+}
+elseif (isset($_POST['clear']) && $_POST['clear']!=='') {
+  $action = 'adminpass';
+} else {
+  $action = 'default';
+}
+
+// Main action switch
+switch ($action) {
+  case 'allusers':
+    // Default value
+    $success = TRUE;
+
+    // Process SpamAssassin
+    if (isset($_POST['on_spamassassin'])) {
+      if ($_POST['on_spamassassin'] != '2' && ($_POST['on_spamassassin'] == '1' || $_POST['on_spamassassin'] == '0')) {
```

Comment:
And there's no point in ensuring that the value is not 2 when the next thing you do is ensure it's either 1 or 0.